Add type column to AuthCredential for STI

brendanshean opened the pull request:
## WHAT
1.  Add type column to AuthCredential in preparation for STI
2.  Add backfill rake task to be run after migration
3.  Remove non-null requirement on AuthCredential provider

## WHY
1. This column will be needed to differentiate subclasses of AuthCredential
2.  Existing records will need this value set for STI to function properly
3.  This will allow for migration removing of provider later

## HOW
1.  Create a Rails migration with: `add_column :auth_credentials, :type, :string`
2.  Iterate through each of the providers and update the appropriate records
3. Add a migration with `change_null_column :auth_credentials, :provider, true`

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No.  Migration file and manual testing of rake task
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

brendanshean committed:
Add backfill to be run after migration

brendanshean committed:
Add type column to AuthCredential for STI

dandrabik reviewed (approved):
👍 Approved, thanks for putting this out first. We chatted about an additional take or leave item, but either way this looks good to me.

brendanshean committed:
Disable STI for now

brendanshean committed:
Fix backfill script

brendanshean committed:
Update rake task

brendanshean committed:
Remove non-null requirement for provider on AuthCredential

brendanshean committed:
Add in_batches to speed up rake task

brendanshean commented:
@anathomical @dandrabik Re-requesting since I added a migration to facilitate a subsequent PR where I remove provider from the schema.  I also updated the backfill script to use `in_batches` and `update_all` for performance.

dandrabik reviewed (approved):
:+1: This looks good
